app/controllers/jobs_controller.rb

```diff
@@ -1,34 +1,85 @@
 # frozen_string_literal: true
 
 class JobsController < ApplicationController
-  skip_before_action :verify_authenticity_token, only: :update_multiple
-
   def index
     return order_jobs if params[:job_ids]
     @date = params[:job_date] || l(Date.current + Settings::Job.lead_time.days)
-    @grouped_jobs = Job.includes({ issue: :category }, %i[group]).order('issue.created_at').group_by_user_group(@date)
+    @grouped_jobs = Job.includes({ issue: :category }, %i[group]).order(order: :asc).group_by_user_group(@date)
   end
 
   def update_multiple
-    @jobs = Job.where(id: params[:job_ids])
-    @group_id = @jobs.first.group.to_s_html
+    @jobs = Job.where(id: params[:job_ids]).order(:order)
+    date = @jobs.first.date
     @jobs.update params.require(:job).permit(:status, :date)
-    render :destroy
+    return reload_all(date) if params[:job][:date].present?
+    render :reload
   end
 
   def destroy
     job = Job.find(params[:id])
     group = job.group
     job.destroy!
-    @group_id = group.to_s_html
-    @jobs = Job.includes({ issue: :category }, %i[group]).where(group_id: group.id, date: params[:date])
+    @jobs = Job.includes({ issue: :category }, %i[group]).where(group_id: group.id, date: params[:date]).order(:order)
+    render :reload
+  end
+
+  def swap_order
+    reorder_jobs(job = Job.find(params[:id]))
+    @jobs = Job.includes({ issue: :category }, %i[group]).where(group_id: job.group.id, date: job.date).order(:order)
+    render :reload
   end
 
   private
 
+  def reload_all(date)
+    @grouped_jobs = Job.includes({ issue: :category }, %i[group]).order(:order).group_by_user_group(date)
+    render :reload_all
+  end
+
   def order_jobs
-    @jobs = Job.includes({ issue: :category }, %i[group]).where(id: params[:job_ids])
-    @goup_id = @jobs.first.group.to_s_html
-    render :destroy
+    order_by_created_at(Job.includes(:issue).where(id: params[:job_ids]).order('issue.created_at ASC'))
+    @jobs = Job.includes({ issue: :category }, %i[group]).where(id: params[:job_ids]).order(:order)
+    render :reload
+  end
+
+  def reorder_jobs(job)
+    prev_job = Job.find_by(id: params[:prev_id])
+    next_job = Job.find_by(id: params[:next_id])
+    return if prev_job.blank? && next_job.blank?
+    return reorder_prev_jobs(job, prev_job) if prev_job && job.order < prev_job.order
+    reorder_next_jobs job, next_job
+  end
+
+  def reorder_prev_jobs(current_job, prev_job)
+    order = current_job.order
+    prev_order = prev_job.order
+    Job.where(job_order_conds(current_job.group_id, order, prev_order + 1)).find_each do |job|
+      job.update(order: job.order - 1)
+    end
+    current_job.update(order: prev_order)
+  end
+
+  def reorder_next_jobs(current_job, next_job)
+    order = current_job.order
+    next_order = next_job.order
```

Comment:
Hier kommt es beim Umsortieren einer Liste mit nur 2 Einträgen derzeit zu einem Laufzeitfehler:
`Started PUT "/jobs/1/swap_order" for 127.0.0.1 at 2021-05-04 08:37:05 +0200
Processing by JobsController#swap_order as JS
  Parameters: {"prev_id"=>"2", "authenticity_token"=>"[filtered]", "id"=>"1"}
  User Load (0.7ms)  SELECT "user".* FROM "user" WHERE "user"."active" = $1 AND "user"."login" ILIKE 'nbennke' LIMIT $2  [["active", true], ["LIMIT", 1]]
  ↳ app/controllers/concerns/authorization.rb:27:in `authenticate'
Username: nbennke
  Job Load (0.6ms)  SELECT "job".* FROM "job" WHERE "job"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
  ↳ app/controllers/jobs_controller.rb:27:in `swap_order'
  Job Load (0.6ms)  SELECT "job".* FROM "job" WHERE "job"."id" = $1 LIMIT $2  [["id", 2], ["LIMIT", 1]]
  ↳ app/controllers/jobs_controller.rb:46:in `reorder_jobs'
  Job Load (0.8ms)  SELECT "job".* FROM "job" WHERE "job"."id" IS NULL LIMIT $1  [["LIMIT", 1]]
  ↳ app/controllers/jobs_controller.rb:47:in `reorder_jobs'
#<NoMethodError: undefined method `order' for nil:NilClass>app/controllers/jobs_controller.rb:64:in `reorder_next_jobs'
 app/controllers/jobs_controller.rb:50:in `reorder_jobs'
 app/controllers/jobs_controller.rb:27:in `swap_order'
`